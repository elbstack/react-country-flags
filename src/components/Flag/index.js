import React, { Component, createElement } from 'react';
import PropTypes from 'prop-types';
import flags from '../../flags/react';

const fetchFlag = async country => {
  if (!flags[country]) return Promise.resolve(null);
  const { default: data } = await flags[country]();
  return data;
};

export const countries = (() => Object.keys(flags))();

export default class Flags extends Component {
  state = {
    flag: null,
  };

  static defaultProps = {
    country: null,
    asSquare: false,
  };

  componentDidMount() {
    this.updateFlag();
  }

  componentDidUpdate(prevProps) {
    const { country } = this.props;

    if (country !== prevProps.country) {
      this.updateFlag();
    }
  }

  async updateFlag() {
    const { country } = this.props;
    const flag = await fetchFlag(country);
    this.setState({ flag });
  }

  render() {
    const { asSquare } = this.props;
    const { flag } = this.state;
    if (!flag) return null;
    if (!asSquare) return createElement(flag);
    return (
      <svg viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
        {createElement(flag)}
      </svg>
    );
  }
}

Flags.propTypes = {
  country: PropTypes.string,
  asSquare: PropTypes.bool,
};
