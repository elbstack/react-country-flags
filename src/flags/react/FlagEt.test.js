import React from 'react';
import renderer from 'react-test-renderer';
import Flag from './FlagEt';

it('FlagEt renders without crashing', () => {
  const component = renderer.create(<Flag />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
