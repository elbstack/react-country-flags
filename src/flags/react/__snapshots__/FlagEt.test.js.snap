// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagEt renders without crashing 1`] = `
<svg
  viewBox="0 0 720 360"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <path
    d="M0 0h720v360H0z"
    fill="#da121a"
  />
  <path
    d="M0 0h720v240H0z"
    fill="#fcdd09"
  />
  <path
    d="M0 0h720v120H0z"
    fill="#078930"
  />
  <g
    transform="translate(360 180)"
  >
    <circle
      fill="#0f47af"
      r={120}
    />
    <g
      id="se519047c"
    >
      <path
        d="M0-96l-4.206 12.944 17.348 53.39h-23.13l-2.599 8h74.163l11.011-8H21.553z"
        fill="#fcdd09"
      />
      <path
        d="M25.863-35.597l30.564-42.069"
        stroke="#fcdd09"
        strokeWidth={4}
      />
    </g>
    <use
      transform="rotate(72)"
      xlinkHref="#se519047c"
    />
    <use
      transform="rotate(144)"
      xlinkHref="#se519047c"
    />
    <use
      transform="rotate(216)"
      xlinkHref="#se519047c"
    />
    <use
      transform="rotate(288)"
      xlinkHref="#se519047c"
    />
  </g>
</svg>
`;
